easybuild/easyblocks/b/boost.py

```diff
@@ -240,7 +240,11 @@ def sanity_check_step(self):
         if self.cfg['boost_mpi']:
             custom_paths["files"].append('lib/libboost_mpi.%s' % shlib_ext)
         if get_software_root('Python'):
-            custom_paths["files"].append('lib/libboost_python.%s' % shlib_ext)
+            pymajorver = get_software_version('Python').split('.')[0]
+            if int(pymajorver) >= 3:
+                custom_paths["files"].append('lib/libboost_python%s.%s' % (pymajorver, shlib_ext))
+            else:
+                custom_paths["files"].append('lib/libboost_python.%s' % shlib_ext)
```

Comment:
To avoid copy-pasting, I'd prefer this:

```python
pymajorver = get_software_version('Python').split('.')[0]
if int(pymajorver) >= 3:
    suff = pymajorver
else:
    suff = ''
custom_paths["files"].append('lib/libboost_python%s.%s' % (suff, shlib_ext))